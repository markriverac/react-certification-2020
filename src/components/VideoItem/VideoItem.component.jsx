import React from 'react';
import './videoItem.styles.css';
import Media from 'react-bootstrap/Media';
import Card from 'react-bootstrap/Card';
import { Link } from 'react-router-dom';

const VideoItem = ({ video }) => {
  const videoId = video.id.videoId || video.id;
  return (
    <Link to={`/video/${videoId}`}>
      <div className="video-item item" aria-hidden="true">
        <Card body style={{ width: '100%' }}>
          <Media as="li">
            <img
              width={350}
              height={180}
              className="mr-3"
              src={video.snippet.thumbnails.medium.url}
              alt={video.snippet.description}
            />
            <Media.Body>
              <div className="video-description">
                <h5 className="video-title">{video.snippet.title}</h5>
                <p className="video-info">{video.snippet.description}</p>
              </div>
            </Media.Body>
          </Media>
        </Card>
      </div>
    </Link>
  );
};
export default VideoItem;
