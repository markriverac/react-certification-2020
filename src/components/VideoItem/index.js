/* istanbul ignore file */
export { default } from './VideoItem.component';
