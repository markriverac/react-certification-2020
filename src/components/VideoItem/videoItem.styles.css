.video-title {
  color: black;
}

.video-info {
  color: rgba(0, 0, 0, 0.8);
}

.video-item {
  display: flex;
  align-items: center !important;
  justify-content: center;
  cursor: pointer;
  text-align: left;
  margin: 20px 0;
}

.video-item.item img {
  max-width: 285px;
  max-height: 250px;
}

.video-description {
  max-height: 170px;
  overflow: hidden;
  text-overflow: ellipsis;
}
