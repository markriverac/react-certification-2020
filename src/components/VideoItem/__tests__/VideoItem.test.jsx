import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';
import VideoItem from '../VideoItem.component';

Enzyme.configure({ adapter: new Adapter() });

it('Renders without exploding', () => {
  const video = {
    id: {
      videoId: 'id',
    },
    snippet: {
      thumbnails: {
        medium: {
          url: 'url',
        },
      },
      title: 'title',
      description: 'description',
    },
  };
  const button = renderer.create(
    <BrowserRouter>
      <VideoItem video={video} />
    </BrowserRouter>
  );
  const tree = button.toJSON();
  expect(tree).toMatchSnapshot();
});
