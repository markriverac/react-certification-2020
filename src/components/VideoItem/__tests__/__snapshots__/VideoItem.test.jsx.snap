// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without exploding 1`] = `
<a
  href="/video/id"
  onClick={[Function]}
>
  <div
    aria-hidden="true"
    className="video-item item"
  >
    <div
      className="card"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        className="card-body"
      >
        <li
          className="media"
        >
          <img
            alt="description"
            className="mr-3"
            height={180}
            src="url"
            width={350}
          />
          <div
            className="media-body"
          >
            <div
              className="video-description"
            >
              <h5
                className="video-title"
              >
                title
              </h5>
              <p
                className="video-info"
              >
                description
              </p>
            </div>
          </div>
        </li>
      </div>
    </div>
  </div>
</a>
`;
